components/nostr-auth.js

```diff
@@ -1,71 +1,78 @@
-import styles from './lightning-auth.module.css'
-import { callWithTimeout } from '@/lib/time'
+import Nostr from '@/lib/nostr'
+import { NDKNip46Signer } from '@nostr-dev-kit/ndk'
+import { useToast } from '@/components/toast'
+import { Button } from 'react-bootstrap'
+import { Form, Input, SubmitButton } from '@/components/form'
+import Moon from '@/svgs/moon-fill.svg'
+import styles from './login.module.css'
 
-function ExtensionError ({ message, details }) {
-  return (
-    <>
-      <h4 className='fw-bold text-danger pb-1'>error: {message}</h4>
-      <div className='text-muted pb-4'>{details}</div>
-    </>
-  )
+const sanitizeURL = (s) => {
+  try {
+    const url = new URL(s)
+    if (url.protocol !== 'https:' && url.protocol !== 'http:') throw new Error('invalid protocol')
+    return url.href
+  } catch (e) {
+    return null
+  }
 }
 
-function NostrExplainer ({ text }) {
+function NostrError ({ message }) {
   return (
     <>
-      <ExtensionError message='nostr extension not found' details='Nostr extensions are the safest way to use your nostr identity on Stacker News.' />
-      <Row className='w-100 text-muted'>
-        <AccordianItem
-          header={`Which extensions can I use to ${(text || 'Login').toLowerCase()} with Nostr?`}
-          show
-          body={
-            <>
-              <Row>
-                <Col>
-                  <ul>
-                    <li>
-                      <a href='https://getalby.com'>Alby</a><br />
-                      available for: chrome, firefox, and safari
-                    </li>
-                    <li>
-                      <a href='https://www.getflamingo.org/'>Flamingo</a><br />
-                      available for: chrome
-                    </li>
-                    <li>
-                      <a href='https://github.com/fiatjaf/nos2x'>nos2x</a><br />
-                      available for: chrome
-                    </li>
-                    <li>
-                      <a href='https://diegogurpegui.com/nos2x-fox/'>nos2x-fox</a><br />
-                      available for: firefox
-                    </li>
-                    <li>
-                      <a href='https://github.com/fiatjaf/horse'>horse</a><br />
-                      available for: chrome<br />
-                      supports hardware signing
-                    </li>
-                  </ul>
-                </Col>
-              </Row>
-            </>
-          }
-        />
-      </Row>
+      <h4 className='fw-bold text-danger pb-1'>error</h4>
+      <div className='text-muted pb-4'>{message}</div>
     </>
   )
 }
 
 export function NostrAuth ({ text, callbackUrl, multiAuth }) {
-  const [createAuth, { data, error }] = useMutation(gql`
+  const [status, setStatus] = useState({
+    msg: '',
+    error: false,
+    loading: false,
+    title: undefined,
+    button: undefined
+  })
+
+  const [suggestion, setSuggestion] = useState(null)
+  const suggestionTimeout = useRef(null)
```

Comment:
The clearSuggestionTimer in the finally block is to avoid causing an useless state update if we are showing the error, the clearSuggestionTimer on unmount is to not hold the context (up to 10 seconds) from the garbage collector.
Generally it is a good practice to make code that clear after itself.